transport/conn.go

```diff
@@ -46,7 +47,7 @@ func (c *connWriter) submit(r request) {
 	c.requestCh <- r
 }
 
-func (c *connWriter) loop() {
+func (c *connWriter) loop(conn *Conn) {
```

Comment:
This looks ugly to me. I wouldn't close connections in their own loops, but rather push error to some higher layer.